watertap/tools/parameter_sweep/parameter_sweep.py

```diff
@@ -841,7 +822,13 @@ def parameter_sweep(
                                 and will not work with future implementations of parallelism.",
                 version="0.10.0",
             )
-
+        # add build functions and kwargs to instance for use with custom function
+        # this might be better to move all of these to Config instead
```

Comment:
I agree

Reply:
I will leave this for another PR (or @k1nshuk  can try to do that in this one, not sure how it would impact parallel implementation as well and another other underlying code (as well as diff_param).